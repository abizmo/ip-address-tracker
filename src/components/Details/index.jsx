import { useSelector } from 'react-redux';
import {
  Card, Detail, DetailContainer, Label,
} from './styles';

function Details() {
  const {
    ip, location, timezone, isp,
  } = useSelector(({ tracker }) => tracker);

  return (
    <Card>
      <DetailContainer>
        <Label>IP Address</Label>
        <Detail>{ip}</Detail>
      </DetailContainer>
      <DetailContainer>
        <Label>Location</Label>
        <Detail>{location}</Detail>
      </DetailContainer>
      <DetailContainer>
        <Label>Timezone</Label>
        <Detail>
          UTC
          {' '}
          {timezone}
        </Detail>
      </DetailContainer>
      <DetailContainer>
        <Label>ISP</Label>
        <Detail>{isp}</Detail>
      </DetailContainer>
    </Card>
  );
}

export default Details;
