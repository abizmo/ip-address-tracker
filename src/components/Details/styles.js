import styled from 'styled-components';

export const Card = styled.div`
  align-items: center;
  background-color: var(--clr-100);
  border-radius: 1rem;
  box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
  padding: 1.5rem;
  position: absolute;
  width: calc(100% - 48px);
`;

export const Label = styled.h2`
  color: var(--clr-400);
  font-size: var(--fs-100);
  font-weight: 700;
  letter-spacing: 0.16em;
  margin-bottom: .5rem;
  text-align: center;
  text-transform: uppercase;
`;

export const Detail = styled.p`
  font-size: var(--fs-500);
  font-weight: 500;
  letter-spacing: -0.01em;
  text-align: center;
`;
