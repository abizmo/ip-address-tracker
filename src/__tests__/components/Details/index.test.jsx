import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Details from '../../../components/Details';

const initialState = {
  tracker: {
    ip: '192.212.174.101',
    location: 'Brooklyn, NY 10001',
    timezone: '-05:00',
    isp: 'SpaceX Starlink',
  },
};

const mockStore = createMockStore([thunk]);

describe('Details test', () => {
  test('should render and match', () => {
    const store = mockStore(initialState);

    const wrapper = mount(
      <Provider store={store}>
        <Details />
      </Provider>,
    );
    const titles = wrapper.find('h2').children();

    expect(wrapper).toMatchSnapshot();
    expect(titles).toHaveLength(4);
    expect(titles.first().text()).toBe('IP Address');
  });
});
