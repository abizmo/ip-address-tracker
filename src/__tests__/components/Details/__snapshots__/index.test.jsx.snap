// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details test should render and match 1`] = `
<div
  className="sc-bdvvtL aZzYl"
>
  <div
    className="sc-gsDKAQ efJMhT"
  >
    <h2
      className="sc-dkPtRN cIkiZQ"
    >
      IP Address
    </h2>
    <p
      className="sc-hKwDye bXAloP"
    >
      192.212.174.101
    </p>
  </div>
  <div
    className="sc-gsDKAQ efJMhT"
  >
    <h2
      className="sc-dkPtRN cIkiZQ"
    >
      Location
    </h2>
    <p
      className="sc-hKwDye bXAloP"
    >
      Brooklyn, NY 10001
    </p>
  </div>
  <div
    className="sc-gsDKAQ efJMhT"
  >
    <h2
      className="sc-dkPtRN cIkiZQ"
    >
      Timezone
    </h2>
    <p
      className="sc-hKwDye bXAloP"
    >
      UTC
       
      -05:00
    </p>
  </div>
  <div
    className="sc-gsDKAQ efJMhT"
  >
    <h2
      className="sc-dkPtRN cIkiZQ"
    >
      ISP
    </h2>
    <p
      className="sc-hKwDye bXAloP"
    >
      SpaceX Starlink
    </p>
  </div>
</div>
`;
